### PR TITLE
Tests - Reposync overhaul

### LINKED ISSUE #3166: Reposync tests are flaky
### Describe the bug

We have wonderful `cobbler reposync` tests thanks to @tpw56j! However in our CI they are flaky.

### Steps to reproduce

1. Look at the history of our CI testruns
2. See that many failed runs are due to the mirrors not being available

### Expected behavior

The Python Tests are thought as Unit Tests. Thus they should be refactored and we should test the integration tests with our bash testsuite.

### Cobbler version

````paste below
Cobbler 3.4.0
  source: 0e20f01b, Mon Jun 13 16:13:33 2022 +0200
  build time: Sun Jul  3 07:59:48 2022
````

### Operating system

openSUSE Leap/Tumbleweed

### Cobbler log

````paste below
Not applicable
````

### Screenshots

None

### Additional information

None

### PR DESCRIPTION
**Pull request by SchoolGuy**
Fixes #3166

### CODE REVIEW

File: tests/actions/reposync_test.py [hunk 1]
```diff
@@ -1,256 +1,525 @@
...
+    with expected_exception:
+        reposync_object.sync(test_repo)
+
+        # Assert
+        call_count = (
+            rhn_sync_mock.call_count
+            + yum_sync_mock.call_count
+            + apt_sync_mock.call_count
+            + rsync_sync_mock.call_count
+            + wget_sync_mock.call_count
```
Comment (agraul) [inline tests/actions/reposync_test.py:470]: nitpick: might be a candidate for `sum()`

**Review by agraul — APPROVED** [review]
Looks good overall, only one test is not clear to me.

File: cobbler/actions/reposync.py [hunk 2]
```diff
@@ -134,13 +136,14 @@ def run(self, name: Optional[str] = None, verbose: bool = True):
...
             # Which may actually NOT reposync if the repo is set to not mirror locally but that's a technicality.
 
+            success = False
             for x in range(self.tries + 1, 1, -1):
-                success = False
                 try:
                     self.sync(repo)
                     success = True
                     break
                 except:
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply cobbler/actions/reposync.py:145]: I can only add `Exception` here and I couldn't log anything specific due to the `utils.log_exc()` function that is used. So I would like to touch it but I think it is better to touch this repo wide for the complete pattern when we are at that point. Still I am going to add `Exception`, doesn't do much but it is something.

File: cobbler/actions/reposync.py [hunk 3]
```diff
@@ -496,7 +498,9 @@ def rhn_sync(self, repo):
 
     # ====================================================================================
 
-    def gen_urlgrab_ssl_opts(self, yumopts) -> Union[str, bool]:
+    def gen_urlgrab_ssl_opts(
+        self, yumopts: Dict[str, Any]
+    ) -> Tuple[Optional[Tuple], bool]:
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply cobbler/actions/reposync.py:503]: Thanks for the suggestion. I will do that another time though. :)

**Pull request merged**